File: src/main/java/mods/cybercat/gigeresque/client/config/ModMenuGigeresque.java
package mods.cybercat.gigeresque.client.config;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;

import eu.midnightdust.lib.config.MidnightConfig;
import mods.cybercat.gigeresque.common.Gigeresque;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
public class ModMenuGigeresque implements ModMenuApi {
	@Override
	public ConfigScreenFactory<?> getModConfigScreenFactory() {
		return parent -> MidnightConfig.getScreen(parent, Gigeresque.MOD_ID);
	}
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/config/ModMenuGigeresque.kt
package com.bvanseg.gigeresque.client.config

import com.bvanseg.gigeresque.common.config.GigeresqueConfig
import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import me.shedaniel.autoconfig.AutoConfig

import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment

/**
 * @author Boston Vanseghi
 */
@Environment(EnvType.CLIENT)
class ModMenuGigeresque : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return ConfigScreenFactory { parent ->
            AutoConfig.getConfigScreen(GigeresqueConfig::class.java, parent).get()
        }
    }
}